components/director/internal/open_resource_discovery/service.go

```diff
@@ -88,6 +94,26 @@ func (s *Service) listAppPage(ctx context.Context, pageSize int, cursor string)
 	if err != nil {
 		return nil, err
 	}
+
+	applicationIDs := make([]string, 0, page.TotalCount)
+	for _, app := range page.Data {
+		applicationIDs = append(applicationIDs, app.ID)
+	}
+
+	labels, err := s.labelRepo.ListGlobalByKeyAndObjects(ctx, model.ApplicationLabelableObject, applicationIDs, applicationTypeLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	appLabelsMap := make(map[string]interface{}, page.TotalCount)
+	for i := range labels {
+		appLabelsMap[labels[i].ObjectID] = labels[i].Value
+	}
+
+	for i, _ := range page.Data {
+		page.Data[i].Labels = []byte(fmt.Sprintf(`{"%s": "%s"}`, applicationTypeLabel, appLabelsMap[page.Data[i].ID]))
```

Comment:
Those are the ORD Labels and not CMP Lables.
Currently, it is safe to override them because we are not updating the app when we fetch the systemInstance property from the document. However, I think it is better not to mess with them as we will do it in the future probably.

I think a better approach (and easier considering this JSON manipulations) is to add `applicatioType` field in the system which is `json:"-"` ignored and use it as a temporary metadata field for this flow.